obp/ope/classification_model.py

```diff
@@ -0,0 +1,660 @@
+
+    fitting_method: str, default='sample'
+        Method to fit the classification model.
+        Must be one of ['sample', 'raw']. Each method is defined as follows:
+            - sample: actions are sampled by applying Gumbel-softmax trick to action_dist_at_position, and action features are represented by one-hot encoding of the sampled action.
+            - raw: action_dist_at_position are directly encoded as action features.
+        If fitting_method is 'raw', action_context must be None.
+
+    calibration_cv: int, default=2
+        Number of folds in the calibration procedure.
+        If calibration_cv <= 1, classification model is not calibrated.
+
+    References
+    -----------
+    Arjun Sondhi, David Arbour, and Drew Dimmery
+    "Balanced Off-Policy Evaluation in General Action Spaces.", 2020.
+
+    """
+
+    base_model: BaseEstimator
+    n_actions: int
+    len_list: int = 1
+    action_context: Optional[np.ndarray] = None
+    fitting_method: str = "sample"
+    calibration_cv: int = 2
+
+    def __post_init__(self) -> None:
+        """Initialize Class."""
+        check_scalar(self.n_actions, "n_actions", int, min_val=2)
+        check_scalar(self.len_list, "len_list", int, min_val=1)
+        check_scalar(self.calibration_cv, "calibration_cv", int)
+        if not (
+            isinstance(self.fitting_method, str)
+            and self.fitting_method in ["sample", "raw"]
+        ):
+            raise ValueError(
+                f"fitting_method must be either 'sample' or 'raw', but {self.fitting_method} is given"
+            )
+        if self.fitting_method == "raw" and self.action_context is not None:
+            check_array(
+                array=self.action_context, name="action_context", expected_dim=2
+            )
+            if (
+                self.action_context.shape
+                != (
+                    self.n_actions,
+                    self.n_actions,
+                )
+                or not np.allclose(self.action_context, np.eye(self.n_actions))
+            ):
+                raise ValueError(
+                    "If fitting_method == 'raw', action_context must be None or identity matrix of size (n_actions, n_actions)."
+                )
+        if not isinstance(self.base_model, BaseEstimator):
+            raise ValueError(
+                "base_model must be BaseEstimator or a child class of BaseEstimator"
+            )
+
+        if self.calibration_cv > 1:
+            self.base_model_list = [
+                clone(
+                    CalibratedClassifierCV(
+                        base_estimator=self.base_model, cv=self.calibration_cv
+                    ),
+                )
+                for _ in np.arange(self.len_list)
+            ]
+        else:
+            self.base_model_list = [
+                clone(self.base_model) for _ in np.arange(self.len_list)
+            ]
+        if self.action_context is None:
```

Comment:
```suggestion
        if self.action_context is None or self.fitting_method == "raw":
```

Reply:
@fullflu I thought we can just set one-hot encoding as action context when self.fitting_method == "raw".